tests/scripts/create_subset_ebas.py

```diff
@@ -0,0 +1,288 @@
+            files_valid.append(file)
+    return ok, files_valid, files_invalid
+
+
+def get_files_var_statlist(data, var, statlist, overlap_found):
+    files = {}
+    for stat in statlist:
+        meta_match = data.find_station_meta_indices(stat)
+
+        for i, idx in enumerate(meta_match):
+            try:
+                sd = data.to_station_data(idx, var)
+            except pya.exceptions.DataCoverageError:
+                continue
+            if len(sd[var].dropna()) > 10:
+                fname = sd.filename
+                if ";" in fname:
+                    raise ValueError("FATAL", var, stat)
+                files[sd.station_name] = [sd.filename]
+                break
+
+        if not overlap_found:
+            for left, right in zip(meta_match[:-1], meta_match[1:]):
+                try:
+                    sdl = data.to_station_data(left, var)
+                    sdr = data.to_station_data(right, var)
+                except pya.exceptions.DataCoverageError:
+                    continue
+                f0, f1 = sdl.filename, sdr.filename
+                merged = sdl.merge_other(sdr, var)
+                if var in merged.overlap:
+                    files[sdl.station_name] = [f0, f1]
+                    overlap_found = True
+    return files, overlap_found
+
+
+def get_files_var_statnum(data, var, statnum):
+    files = {}
+    numfound = 0
+    allstats = data.unique_station_names
+    for stat in allstats:
+        if numfound == statnum:
+            break
+        meta_match = data.find_station_meta_indices(stat)
+
+        for i, idx in enumerate(meta_match):
+            try:
+                sd = data.to_station_data(idx, var)
+            except pya.exceptions.DataCoverageError:
+                continue
+            if len(sd[var].dropna()) > 10:
+                fname = sd.filename
+                if ";" in fname:
+                    raise ValueError("FATAL", var, stat)
+                files[sd.station_name] = [sd.filename]
+                numfound += 1
+                break
+    return files
+
+
+def main():
+
+    # reader = pya.io.ReadUngridded(NAME, data_dir=EBAS_BASE_DIR)
+    reader = pya.io.ReadUngridded(
+        NAME,
+    )
+    r_lowlev = reader.get_lowlevel_reader(NAME)
+
+    # r_lowlev._dataset_path = ebas_local
+
+    # directory containing NASA Ames files
+    FILEDIR_SRC = r_lowlev.file_dir
+    print(f"checking for invalid files in static file list...")
+    CURRENT_OK, files_valid, files_invalid = check_outdated(FILEDIR_SRC)
+    print(f"done...")
+
+    if not CURRENT_OK:
+        print(f"outdated files: {files_invalid}")
+        print("---------------------WARNING------------------------------")
+        print(
+            "Some files in current EBAS subset do not exist anymore in "
+            "the most recent EBAS database, consider updating them"
+        )
+        print("---------------------WARNING------------------------------")
+        raise Exception
```

Comment:
Is this an exception or a warning?
I think this should be

``` python
    if not CURRENT_OK:
        raise RuntimeError(
            "Some files in current EBAS subset do not exist anymore in "
            "the most recent EBAS database. "
            f"Outdated files: {files_invalid}"
        )
```

Reply:
I put that into a separate issue #604. A bit out of scope for now.